### ISSUE TITLE
[no-empty-interface] can't extend Array, "declaring no members is equivalent to its supertype"

### ISSUE THREAD

**Issue by David-Else** [typescript-eslint/typescript-eslint#667]
```JSON
rules = recommended
```

```TS
interface Hiscore {
  name: string;
  score: number;
}

interface Hiscores extends Array<Hiscore> {}
```

**Expected Result**
no error
**Actual Result**
`An interface declaring no members is equivalent to its supertype. [Error/@typescript-eslint/no-empty-interface]`
**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.11.0` |
| `@typescript-eslint/parser`        | `1.11.0` |
| `TypeScript`                       | `X.Y.Z` |
| `ESLint`                           | `6.0.1` |
| `node`                             | `12.3.1` |
| `npm`                              | `X.Y.Z` |

**Comment by bradzacher** [comment]
the solution to this is

```ts
type Hiscores = Array<Hiscore>
```

**Comment by David-Else** [comment]
@bradzacher Cheers! Eslint did not like that, but likes `type Hiscores = Hiscore[];` which seems even nicer :)
Out of interest, do you consider this issue a bug? It would be nice if it automatically suggested your solution.

**Comment by bradzacher** [comment]
I hadn't thought of providing a fixer for this.
We've had this issue raised a couple of times, so it's probably worth adding a fixer for this specific message - it's a pretty straightforward fix!

**Comment by dimabory** [comment]
@bradzacher Are you still fine with adding a fixer "noEmptyWithSuper" message? I'm gonna take it over and open PR.

**Comment by bradzacher** [comment]
Definitely, go for it!
This is a common issue that people have raised a few times, so a fixer for it would be ace.
